mito-ai/src/Extensions/AiChat/ChatTaskpane.tsx

```diff
@@ -94,68 +115,143 @@ const ChatTaskpane: React.FC<IChatTaskpaneProps> = ({
+      const metadata: IGetThreadsMetadata = {
+        promptType: "get_threads"
+      };
+
+      const chatThreadsResponse = await websocketClient.sendMessage<
+        ICompletionRequest, 
+        IFetchThreadsReply
+      >({
+         type: "get_threads",
+         message_id: UUID.uuid4(),
+         metadata: metadata,
+         stream: false
+      });
+
+      setChatThreads(chatThreadsResponse.threads);
+    };
+
+    const fetchChatHistoryForThread = async (threadId: string) => {
+      await websocketClient.ready;
+
+      const metadata: IFetchHistoryMetadata = {
+        promptType: "fetch_history",
+        threadID: threadId
+      };
+
+      const fetchHistoryCompletionRequest: IFetchHistoryCompletionRequest = {
+        type: 'fetch_history',
+        message_id: UUID.uuid4(),
+        metadata: metadata,
+        stream: false
+    }
+
+      const chatHistoryResponse = await websocketClient.sendMessage<
+        ICompletionRequest, 
+        IFetchHistoryReply
+      >(fetchHistoryCompletionRequest);
+
+      // Create a fresh ChatHistoryManager and add the initial messages
+      const newChatHistoryManager = getDefaultChatHistoryManager(
+        notebookTracker,
+        contextManager
+    );
+
+      // Add messages to the ChatHistoryManager
+      chatHistoryResponse.items.forEach(item => {
+        try {
+            // If the user sent a message in agent mode, the ai response will be a JSON object
+            // which we need to parse. 
+            const agentResponse = JSON.parse(item.content as string);
+            handleAgentResponse(agentResponse, newChatHistoryManager);
+        } catch {
+            newChatHistoryManager.addChatMessageFromHistory(item);
         }
-        
-        const chatHistoryResponse = await websocketClient.sendMessage(fetchHistoryCompletionRequest);
+      });
 
-        return chatHistoryResponse.items.map((item: any) => ({
-            role: item.role,
-            content: item.content
-        }));
+      // Update the state with the new ChatHistoryManager
+      setChatHistoryManager(newChatHistoryManager);
+      setActiveThreadId(threadId);
+    };
+    
+    const deleteThread = async (threadId: string) => {
+      await websocketClient.ready;
+
+      const metadata: IDeleteThreadMetadata = {
+        promptType: "delete_thread",
+        threadID: threadId
+      };
+
+      const response = await websocketClient.sendMessage<
+        ICompletionRequest, 
+        IDeleteThreadReply
+      >({
+         type: "delete_thread",
+         message_id: UUID.uuid4(),
+         metadata: metadata,
+         stream: false
+      });
+
+      if(response.created === true) {
+         const updatedThreads = chatThreads.filter(thread => thread.thread_id !== threadId);
+         setChatThreads(updatedThreads);
+         if(activeThreadId === threadId) {
+           if(updatedThreads.length > 0) {
+              fetchChatHistoryForThread(updatedThreads[0].thread_id);
```

Comment:
I thought this was a bug at first because the active chat doesn't always need to be the first chat in the thread, but then I noticed that when we change to a different chat, we make that the active chat. 

So in that case, do we even need to separately keep track of the `activeThreadId`? Can't we just use the first chat always?